tests/services/base/conftest.py

```diff
@@ -0,0 +1,76 @@
+from inspect import getmembers, isclass
+from types import ModuleType
+
+import pytest
+from airflow.models.connection import Connection
+
+from orca.services import base, nextflowtower, sevenbridges
+
+SERVICES = {
+    nextflowtower: "tower://:foo@api.tower.nf/?workspace=bar/baz",
+    sevenbridges: "sbg://:foo@api.sbgenomics.com/v2/?project=bar/baz",
+}
+
+
+def get_service_name(service_module: ModuleType) -> str:
+    return service_module.__name__.split(".")[-1]
+
+
+@pytest.fixture(params=SERVICES, ids=get_service_name)
```

Comment:
TIL: you can pass in a function in the `ids`

This is an elegant solution to run the tests across multiple service. Nice!

Nit: We may want to add in the CONTRIBUTING guide to add to the `SERVICES` dict when new services are added to add to the test

Reply:
Good idea! Instead of documenting this, I added a test to make sure that all services are listed in that dictionary (`test_that_all_implemented_services_are_being_tested`). This way, it can be enforced automatically.